### DISCUSSION TITLE
SCM with active LSM generates surface turbulent fluxes that are off compared with prescribed surface fluxes

### DISCUSSION THREAD

**Discussion by XiaSun-Atmos** [NCAR/ccpp-scm#369] [Cases, Initial Conditions, Forcing] [answered]
Hi,
I am using the UFS_IC_generator.py to generate the forcing files for CCPP v5. However, I found that with active LSM in SCM, the magnitude of simulated surface fluxes is off compared with the prescribed surface fluxes. Did I miss something?
Thanks!

SCM output with prescribed surface flux:
<img width="1319" alt="Screen Shot 2023-03-21 at 1 08 18 PM" src="https://user-images.githubusercontent.com/58949533/226715577-9fd99692-0574-4c34-92d9-34934354b12e.png">

SCM output with active LSM:
<img width="961" alt="Screen Shot 2023-03-21 at 1 09 24 PM" src="https://user-images.githubusercontent.com/58949533/226715726-a86163e3-6661-4c73-9040-460b3c714f3b.png">

**Comment by grantfirl** [comment]
Hi Xia,

Where do the prescribed surface flux data come from? Are we comparing apples-to-apples? I would certainly expect the LSM to produce different surface fluxes than, say, observations, right?

-Grant

**Comment by XiaSun-Atmos** [comment]
Thanks @grantfirl . The prescribed surface fluxes are from 3-D UFS runs. The soil properties are extracted from 3-D UFS runs to drive SCM with active LSM. I guess the surface fluxes are not exactly apples-to-apples, since one is from 3D model and the other is from SCM with active LSM. So under this condition, would you say the differences between the two scenarios look reasonable?
Thanks,

**Comment by grantfirl** [comment]
I would say that the differences are too large to be reasonable in that case. There are a couple of things to note, however. First, I believe that for v5, the UFS_IC_generator script is calculating ICs only, not forcing, so I would be careful to interpret anything beyond a simulation of an hour or so. Second, the prescribed sensible heat flux is into the surface quite a bit during the simulation, denoting a colder surface than the surface layer. Since the LSM results aren't showing this, what is happening to the surface temperature in the active LSM run?

**Comment by ligiabernardet** [comment]
@XiaSun-Atmos I am following up on this. Do you have want to continue exploring this problem?

**Comment by XiaSun-Atmos** [comment]
@grantfirl This is what the surface temperature looks like. It seems to capture the diurnal variations well. 
<img width="655" alt="Screen Shot 2023-03-22 at 12 26 56 PM" src="https://user-images.githubusercontent.com/58949533/230466373-2b4e503f-99d8-43c4-99c0-23c46c781b5c.png">

**Comment by dustinswales** [comment]
@XiaSun-Atmos Recently a new feature to run the SCM using UFS ICs and advective forcing has become available, see section 5.5 of the [Users Guide](https://github.com/NCAR/ccpp-scm/blob/main/scm/doc/TechGuide/main.pdf). There is the option to create a UFS reference file for comparison with SCM output, which should give you a more apples-to-apples comparison than cases that use prescribed surface forcing.

**Comment by XiaSun-Atmos** [comment]
Thank you @dustinswales ! I will give it a try.

**Reply by XiaSun-Atmos** [reply]
@dustinswales I tried the ufs replay. Thank you for adding this feature.

comparison data:
![image](https://github.com/NCAR/ccpp-scm/assets/58949533/9ae44621-f3e7-479e-bb09-57a9123b7c97)


scm output:
![image](https://github.com/NCAR/ccpp-scm/assets/58949533/9093a4a7-0614-4cee-9619-5b4bbde1395d)

Do you think is it possible that there might be some unit/conversion issues when calculating shf and lhf in the SCM code?
Thanks,

**Comment by ligiabernardet** [comment]
@dustinswales @hertneky Can you please help @XiaSun-Atmos resolve this conundrum? The issue is that the SCM (with active LSM) initialized and forced from UFS fields produces results that are too different in magnitude from the UFS itself. This differs from expectations. Why would differences be so big?

**Comment by dustinswales** [comment]
@XiaSun-Atmos @ligiabernardet 
I plan on getting to this soon. I've been juggling too many tasks lately and haven't gotten around to this. I'll earmark some time for this Friday.
Unless @hertneky wants to take a look?

**Comment by dustinswales** [comment]
@XiaSun-Atmos Sorry for the delay, I'm looking at this now. A couple quick questions.
Can you share with me the UFS data you are using to generate the replay case? Either on Hera or Cheyenne?
Along with that, the commands you are using to call the scripts? (e.g. the locations your are trying to extract)

**Reply by XiaSun-Atmos** [reply]
@dustinswales Thanks you for taking a look at this! I sent you the directory paths offline.

**Comment by dustinswales** [comment, marked as answer]
@XiaSun-Atmos Alright, I think I have it working now. My apologizes for the latency. I opened a PR with the fix.

surface sensible and latent heat fluxes for SCM UFS-replay for three gridpoints [Top: 262.25, 36.40] [Mid: 262.50, 36.60] [Bot:262.75, 36.80]
![Screenshot 2023-08-29 at 2 53 20 PM](https://github.com/NCAR/ccpp-scm/assets/13949033/63c54aee-0316-4289-a575-a51904070131)
surface sensible and latent heat fluxes for UFS output
![Screenshot 2023-08-29 at 2 53 27 PM](https://github.com/NCAR/ccpp-scm/assets/13949033/90d4f0f0-b3b6-409a-9937-744a6bb73257)

**Reply by dustinswales** [reply]
@XiaSun-Atmos @ligiabernardet A quick note.
There are still some differences between the SCM-replay and the UFS output, but recall that the replay script uses the UFS history files to create the forcing, not the dynamic tendencies (diagnostic). That's just to say that the impact of the physics is already baked into the forcing, so it's not "pure" replay as if we were using the dynamical tendencies.

This was done to be able to use the replay feature on any UFS output, w/o needing to rerun the case with extra diagnostics being created. So I'm not surprised that they are not identical, but I don't have a good sense for how different they should be. I think they are much more reasonable with the fix than before, but not perfect.

**Reply by XiaSun-Atmos** [reply]
Thanks @dustinswales for identifying and fixing the bug so quickly! It absolutely looks much better than the earlier version.
I agree with you there are still differences (some are near 28%, some are very close) compared with ufs output. It might also be possible there are some clouds formed in SCM, but not in UFS. I will think about it more. Thanks!
